arangod/Cache/Table.cpp

```diff
@@ -178,72 +254,65 @@ void* Table::primaryBucket(uint64_t index) {
   return &(_buckets[index]);
 }
 
-std::unique_ptr<Table::Subtable> Table::auxiliaryBuckets(uint32_t index) {
+std::unique_ptr<Table::Subtable> Table::auxiliaryBuckets(std::uint32_t index) {
   if (!isEnabled()) {
     return std::unique_ptr<Subtable>(nullptr);
   }
   GenericBucket* base;
-  uint64_t size;
-  uint32_t mask;
-  uint32_t shift;
+  std::uint64_t size;
+  std::uint32_t mask;
+  std::uint32_t shift;
 
-  _lock.lockRead();
+  SpinLocker guard(SpinLocker::Mode::Read, _lock);
   std::shared_ptr<Table> source = _auxiliary->shared_from_this();
   TRI_ASSERT(_auxiliary.get() != nullptr);
   if (_logSize > _auxiliary->_logSize) {
-    uint32_t diff = _logSize - _auxiliary->_logSize;
+    std::uint32_t diff = _logSize - _auxiliary->_logSize;
     base = &(_auxiliary->_buckets[index >> diff]);
     size = 1;
     mask = 0;
     shift = 0;
   } else {
-    uint32_t diff = _auxiliary->_logSize - _logSize;
+    std::uint32_t diff = _auxiliary->_logSize - _logSize;
     base = &(_auxiliary->_buckets[index << diff]);
-    size = static_cast<uint64_t>(1) << diff;
-    mask = (static_cast<uint32_t>(size - 1) << _auxiliary->_shift);
+    size = static_cast<std::uint64_t>(1) << diff;
+    mask = (static_cast<std::uint32_t>(size - 1) << _auxiliary->_shift);
     shift = _auxiliary->_shift;
   }
-  _lock.unlockRead();
 
   return std::make_unique<Subtable>(source, base, size, mask, shift);
```

Comment:
In contrast to the previous implementation, we are now holding the spin-lock while the `make_unique` is executing.
It would be a small performance optimization to call `make_unique` only after the lock has been released.